server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -408,7 +408,10 @@ public short partitionRecordVersion() {
     }
 
     public short fetchRequestVersion() {
-        if (this.isAtLeast(IBP_3_7_IV4)) {
+        // TODO: update this to 3.9 when it is available
+        if (this.isAtLeast(IBP_4_0_IV0)) {
```

Comment:
TODO reminder. Change this to 3.9 when it is available. Should we create it before merging this PR?

Reply:
I think as long as we can remember to do it, it's fine we create a JIRA ticket and do this later.

Reply:
The issue is that this test is failing: `kafka.server.ReplicaFetcherThreadTest.shouldSendLatestRequestVersionsByDefault`

That test assumes that the replica fetcher thread always sends the latest Fetch request version for the latest production metadata version. I don't think that is correct in all cases. I assume that whomever added the test wanted to make sure that new Fetch version have an associated MetadataVersion.

Let me see if I can figure out a way to fix that test.

Reply:
I fixed the test and filed this issue: [Metadata version 3.9 should return Fetch version 17](https://issues.apache.org/jira/browse/KAFKA-17018)